api/src/main/java/org/apache/iceberg/catalog/ViewCatalog.java

```diff
@@ -0,0 +1,149 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.catalog;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchViewException;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.view.View;
+import org.apache.iceberg.view.ViewBuilder;
+import org.apache.iceberg.view.ViewRepresentation;
+
+/** A Catalog API for view create, drop, and load operations. */
+public interface ViewCatalog {
+
+  /**
+   * Return the name for this catalog.
+   *
+   * @return this catalog's name
+   */
+  String name();
+
+  /**
+   * Return all the identifiers under this namespace.
+   *
+   * @param namespace a namespace
+   * @return a list of identifiers for views
+   * @throws NotFoundException if the namespace is not found
+   */
+  List<TableIdentifier> listViews(Namespace namespace);
+
+  /**
+   * Load a view.
+   *
+   * @param identifier a view identifier
+   * @return instance of {@link View} implementation referred by the identifier
+   * @throws NoSuchViewException if the view does not exist
+   */
+  View loadView(TableIdentifier identifier);
+
+  /**
+   * Check whether view exists.
+   *
+   * @param identifier a view identifier
+   * @return true if the table exists, false otherwise
+   */
+  default boolean viewExists(TableIdentifier identifier) {
+    try {
+      loadView(identifier);
+      return true;
+    } catch (NoSuchViewException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Create a view.
+   *
+   * @param identifier a view identifier
+   * @param schema view schema
+   * @param representation a view representation
+   * @param properties a string map of view properties
+   */
+  default View createView(
+      TableIdentifier identifier,
+      Schema schema,
+      ViewRepresentation representation,
+      Map<String, String> properties) {
+    return buildView(identifier)
+        .withSchema(schema)
+        .withRepresentation(representation)
+        .withProperties(properties)
+        .create();
+  }
+
+  /**
+   * Instantiate a builder to create or replace a view.
+   *
+   * @param identifier a view identifier
+   * @return a view builder
+   */
+  ViewBuilder buildView(TableIdentifier identifier);
+
+  /**
+   * Instantiate a builder to create or replace a view with only SQL view representation.
+   *
+   * @param identifier a view identifier
+   * @return a view builder
+   */
+  ViewBuilder buildSqlView(TableIdentifier identifier);
```

Comment:
I would rather have a single `buildView` method that allows you to create a SQL view. There are no other representations right now so we should focus on SQL. Rather than exposing `ViewRepresentation` right now, we can use methods that update the SQL representation, like `withQuery(String)`.

Reply:
Totally fine with this approach.

- buildView() + create/replace(): create or replace sql view representation
- loadView().updateRepresentations() + commit(): update other representations